Sources/Build/TestMetadataParser.swift

```
@@ -0,0 +1,112 @@
+    /// Returns array because there might be more test classes in one file
+    func parseTestClasses(testFilePath: String) throws -> [TestClassMetadata]? {
+        
+        guard let lines = try? File(path: testFilePath).enumerate() else { return nil }
+        
+        var classes = [TestClassMetadata]()
+        func finishClass(className: String?, _ testNames: [String]) {
+            if let className = className where testNames.count > 0 {
+                let testClass = TestClassMetadata(name: className, testNames: testNames)
+                classes.append(testClass)
+            }
+        }
+        
+        //parse lines one by one, no need to load the whole into memory
+        var className: String? = nil
+        var testNames = [String]()
+        while true {
+            
+            guard let line = lines.next() else {
+                //EOF, no more classes
+                finishClass(className, testNames)
+                testNames = []
+                return classes.nilIfEmpty()
+            }
+            
+            //look for a class name
+            if let newClassName = parseClassName(line) {
+                //found class name, finish old class and start new
+                finishClass(className, testNames)
+                className = newClassName
+                testNames = []
+            }
+            
+            //look for a test case
+            if let testName = parseTestName(line) {
+                //found another test name
+                testNames.append(testName)
+            }
+        }
+    }
+    
+    func parseClassName(line: String) -> String? {
+        var comps = line.splitWithCharactersInString(":\r\t\n ").filter { !$0.isEmpty }
+        guard comps.count >= 2 else { return nil }
+        //see if this line is a candidate
+        guard Set(comps).contains("class") else { return nil }
+        while true {
+            let allowedFirst = Set(["private", "public", "internal", "final"])
```

I'm just wondering whether you think it's an appropriate thing to do - to not let the compiler fail, instead fail earlier in the process with the specific error text. I think it is but I'm looking for a second set of eyes here.